packages/notifications/push/amplify_push_notifications/ios/Classes/AmplifyPushNotificationsPlugin+AppDelegate.swift

```diff
@@ -0,0 +1,89 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Foundation
+import Flutter
+
+private let completionHandlerIdKey = "completionHandlerId"
+
+extension AmplifyPushNotificationsPlugin {
+    public func application(
+        _ application: UIApplication,
+        didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
+    ) {
+        let token = deviceToken.map { String(format: "%02.2hhx", $0) }.joined()
+
+        // the cached token check is to reduce the frequence of emitting TOKEN_RECEIVED events
+        // the deviceToken that client gets is still from the system directly
+        if (cachedDeviceToken != token) {
+            cachedDeviceToken = token
+            EventsStreamHandlers.shared.tokenReceived.sendEvent(payload: ["token": token])
+        }
+    }
+
+    public func application(
+        _ application: UIApplication,
+        didFinishLaunchingWithOptions launchOptions: [AnyHashable : Any] = [:]
+    ) -> Bool {
+        if let remoteNotification = launchOptions[UIApplication.LaunchOptionsKey.remoteNotification] as? [AnyHashable: Any] {
+            // If the App is awaken by a remote notification when background mode is enabled
+            if application.applicationState == .background {
+                isBackgroundMode = true
+            }
+
+            // When the App is launched (from terminated state) by tapping on a notification
+            // This happens whne background mode is not enabled
+            if application.applicationState != .background {
+                launchNotification = remoteNotification
+            }
+        }
+
+        return true
+    }
+
+    public func application(
+        _ application: UIApplication,
+        didFailToRegisterForRemoteNotificationsWithError error: Error
+    ) {
+        print("registerForRemoteNotifications failed with error: \(error).")
```

Comment:
Print on Dart side using `safePrint` - not sure if this is safe to print in prod

Reply:
We are sending the error over to the Dart impl. now this print is unnecessary, removed.